lib/src/models/nodes.ts

```diff
@@ -80,3 +81,34 @@ export interface BodyNode {
   description?: Locatable<string>;
   type: DataType;
 }
+
+export interface TestNode {
+  description?: Locatable<string>;
+  states?: TestStateNode[];
+  request?: Locatable<TestRequestNode>;
+  response: Locatable<TestResponseNode>;
+}
+
+export interface TestStateNode {
+  name: string;
+  params?: { name: string; expression: DataExpression }[];
```

Comment:
Should these (and below in TestRequestNode and TestResponseNode) also not be optional?